### ISSUE TITLE
Some conceptual confusion

### ISSUE THREAD

**Issue by MavenZheng1003** [LARS-research/AdaProp#7]
Hello!
I'm sorry to bother you. Recently, when I read your paper, the cognitive boundary between KBQA task and KB reasoning task is quite blurred. Especially in this work, I think it is highly similar to the two stages of KBQA knowledge retrieval and semantic parsing. Could you please answer that for me, because if it is a KBQA task, I may need to select a data set like CWQ to try later. Looking forward to your reply, thanks again!

**Comment by yzhangee** [comment]
Hi, thank you for your interest. The connection between KGQA and KG reasoning indeed has a strong connection. In practice, KBQA focuses on understanding the query (in text), and then obtains the answer from KG or retrieves relevant knowledge. For KG reasoning, it focuses more on exploring the answer over KG given a query like (e_q, r_q, ?), where the query entity and relation are given. 

Since AdaProp contains no textual understanding and processing module, it cannot be directly used for KGQA tasks. However, with some adaptation, it can be used for the KGQA task by specially processing text information. 

In summary, for the method itself, you do not need to test it on KGQA datasets like CWQ. But if you are interested in adapting it to KGQA, some customized design on text should help.

**Comment by MavenZheng1003** [comment]
I see, can KG reasoning be approximated as a subtask of KBQA? Because KBQA needs to complete the understanding of the question before completing KG reasoning.Thank you very much for your patience again!
